dd-trace-api/src/main/java/datadog/trace/api/Ids.java

```diff
@@ -5,9 +5,31 @@
 
 /** Conversions between DataDog decimal ids and Zipkin hex ids. Supports both 64 and 128 bit ids. */
 public class Ids {
-  private static char[] pad = "00000000000000000000000000000000".toCharArray();
+  private static final char[] pad = "00000000000000000000000000000000".toCharArray();
+  private static final char[] HexChars = {
+    '0', '1', '2', '3',
+    '4', '5', '6', '7',
+    '8', '9', 'a', 'b',
+    'c', 'd', 'e', 'f',
+  };
 
   public static char[] idToHexChars(final String id) {
+    if (id.length() <= 20) {
+      // *likely* (but not definitely) a 64-bit id: try to parse as such
+      try {
+        long val = Long.parseLong(id);
+        char[] answer = new char[16];
+        for (int i = 0; i < 16; i++) {
+          long mask = (0x0FL << (i * 4));
```

Comment:
these can be precomputed or will the jvm likely do that?

Reply:
Not worth it; a move is so much more expensive than a one-op shift.  Also, the memory allocation saved here outweighs either by several orders of magnitude.